ray-operator/controllers/raycluster_controller.go

```diff
@@ -279,9 +308,11 @@ func (r *RayClusterReconciler) reconcilePods(instance *rayiov1alpha1.RayCluster)
 			}
 			instance.Spec.WorkerGroupSpecs[index].ScaleStrategy.WorkersToDelete = []string{}
 			continue
-		} else if *worker.Replicas < int32(len(runningPods.Items)) {
+		} else {
+			// diff < 0 and not the same absolute value as int32(len(worker.ScaleStrategy.WorkersToDelete)
 			// we need to scale down
 			workersToRemove := int32(len(runningPods.Items)) - *worker.Replicas
+			//// SriramQ: Isn't this too early? This does not consider the IsNotFound case (see below)
```

Comment:
in the code block 255-275, you can probably track the number of pods deleted (excluding NOT_FOUND).